production/ksonnet/loki/querier.libsonnet

```diff
@@ -38,7 +38,8 @@ local k = import 'ksonnet-util/kausal.libsonnet';
     ) +
     deployment.mixin.spec.strategy.rollingUpdate.withMaxSurge('15%') +
     deployment.mixin.spec.strategy.rollingUpdate.withMaxUnavailable('15%') +
-    if $._config.querier.use_topology_spread then
+    if $._config.querier.use_no_constraints then {}
```

Comment:
Probably helpful to add a check ensuring `use_topology_stread` is false so users don't see it overridden silently. e.g. "if unconstrained && use_spread {error}"

See errors/asserts in the jsonnet [docs](https://jsonnet.org/learning/tutorial.html)